nixos/modules/services/misc/lms.nix

```diff
@@ -0,0 +1,234 @@
+                docroot = "${pkgs.lms}/share/lms/docroot/;/resources,/css,/images,/js,/favicon.ico";
+                approot = "${pkgs.lms}/share/lms/approot";
+                http-server-thread-count = ${toString cfg.theadCount};
+              ''
+              + strings.optionalString (cfg.listenbrainzApiBaseUrl != null) ''
+                listenbrainz-api-base-url = "${cfg.listenbrainzApiBaseUrl}";
+              ''
+              + ''
+                listenbrainz-max-sync-listen-count = ${toString cfg.listenbrainzMaxSyncListenCount};
+                listenbrainz-sync-listens-period-hours = ${toString cfg.listenbrainzSyncPeriod};
+              ''
+              + strings.optionalString (cfg.acousticbrainzApiBaseUrl != null) ''
+                acousticbrainz-api-base-url = "${cfg.acousticbrainzApiBaseUrl}";
+              ''
+              + ''
+                login-throttler-max-entries = ${toString cfg.loginThrottlerMaxEntries};
+                api-subsonic = ${boolToString cfg.enableSubsonicApi};
+                cover-max-file-size = ${toString cfg.coverArtMaxSize};
+                cover-max-cache-size = ${toString cfg.coverArtCacheSize};
+                cover-jpeg-quality = ${toString cfg.coverArtJpegQuality};
+              '');
+          in "${pkgs.lms}/bin/lms ${lmsConfig}";
+        Restart = "on-failure";
+        RestartSec = 5;
+        User = cfg.user;
+        Group = "lms";
+        WorkingDirectory = cfg.home;
+        UMask = "0022";
+
+        NoNewPrivileges = true;
+        ProtectSystem = true;
+        ProtectHome = true;
+        PrivateTmp = true;
+        PrivateDevices = true;
+        ProtectClock = true;
+        ProtectKernelTunables = true;
+        ProtectKernelModules = true;
+        ProtectKernelLogs = true;
+        ProtectControlGroups = true;
+      };
+    };
+
+    # from: https://github.com/epoupon/lms#reverse-proxy-settings
+    services.nginx = mkIf (cfg.virtualHost != null) {
+      enable = true;
+      virtualHosts.${cfg.virtualHost} = {
+        locations."/" = {
+          proxyPass = "http://${cfg.listenAddress}:${toString cfg.port}";
+          extraConfig = ''
+            proxy_set_header Host $host;
+            proxy_set_header X-Real-IP $remote_addr;
+            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+            proxy_set_header X-Forwarded-Proto $scheme;
+            proxy_read_timeout 120;
+          '';
+        };
+        extraConfig = ''
+          proxy_request_buffering off;
+          proxy_buffering off;
+          proxy_buffer_size 4k;
+          proxy_read_timeout 10m;
+          proxy_send_timeout 10m;
+          keepalive_timeout 10m;
+        '';
+      };
+    };
+
+    users.users = mkIf (cfg.user == "lms") {
+      lms = {
+        isSystemUser = true;
+        description = "LMS service user";
+        name = cfg.user;
+        group = "lms";
+        home = cfg.home;
+        createHome = true;
```

Comment:
This is usually a bad idea: https://github.com/NixOS/nixpkgs/pull/115332#issuecomment-792287827

How about we use this instead:
```nix
systemd.services.lms.serviceConfig = mkMerge [
  {
    ExecStart = ...
    Restart = ...
    User = ...
  }
  (mkIf (cfg.home == "lms") {
    StateDirectory = "lms";
  })
];
```

Then we can change the description of `services.lms.home` to:

The directory where lms uses as a working directory to store its state. If left as the default value this directory will automatically be created before the lms server starts, otherwise the sysadmin is responsible for ensuring the directory exists with appropriate ownership and permissions.

Reply:
Honestly, I didn’t know systemd even had this option, which is exactly what is needed most of the time for a service. Changed.